Downgrade Python to 3.12 due to SSL issues

tomach opened the pull request:
## Summary of changes
This PR downgrades the project’s Python version from 3.13 to 3.12 to work around SSL certificate verification issues with AKS k8s API introduced in Python 3.13. The issue appears to stem from stricter behavior in urllib3, which is pulled in indirectly by dependencies

```
SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Missing Authority Key Identifier
```

## Checklist

- [ ] Link to issue this PR refers to:
- [ ] Relevant changes are reflected in `CHANGES.rst`
- [ ] Added or changed code is covered by tests
- [ ] Documentation has been updated if necessary
- [ ] Changed code does not contain any breaking changes (or this is a major version change)

plaharanne commented on setup.py:
```diff
@@ -90,7 +90,6 @@ def read(path: str) -> str:
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
         "Programming Language :: Python :: 3.12",
-        "Programming Language :: Python :: 3.13",
```
this line was added by this PR https://github.com/crate/crate-operator/commit/21275f8336d6a5731c258289f6cb539c2a608a3c
I don't know if it's necessary to remove it.

tomach replied:
not sure, but I think it's fine to remove it for now since 3.13 is not (yet) supported.

tomach commented:
> In the previous PR, the version was also changed in .readthedocs.yml, line 5. Otherwise, it looks good to me.

thanks, I also updated it in `.readthedocs.yml`
